pkg/kube/controllers/waitservice/pod_mutator.go

```diff
@@ -0,0 +1,130 @@
+package waitservice
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"code.cloudfoundry.org/quarks-operator/pkg/kube/apis"
+	"code.cloudfoundry.org/quarks-operator/pkg/kube/util/operatorimage"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"code.cloudfoundry.org/quarks-utils/pkg/config"
+)
+
+var (
+	// WaitKey is the key for identifying which service the pod has to wait for
+	WaitKey = fmt.Sprintf("%s/wait-for", apis.GroupName)
+)
+
+// PodMutator for adding waiting init containers for linked services
+type PodMutator struct {
+	client  client.Client
+	log     *zap.SugaredLogger
+	config  *config.Config
+	decoder *admission.Decoder
+}
+
+// Check that PodMutator implements the admission.Handler interface
+var _ admission.Handler = &PodMutator{}
+
+// NewPodMutator returns a new mutator that adds waiting InitContainers
+func NewPodMutator(log *zap.SugaredLogger, config *config.Config) admission.Handler {
+	mutatorLog := log.Named("waitservice-pod-mutator")
+	mutatorLog.Info("Creating a Pod mutator for WaitService")
+
+	return &PodMutator{
+		log:    mutatorLog,
+		config: config,
+	}
+}
+
+func validWait(annotations map[string]string) bool {
+	valid := false
+	if annotations[WaitKey] != "" {
+		valid = true
+	}
+
+	return valid
+}
```

Comment:
```suggestion
func validWait(annotations map[string]string) bool {
	return annotations[WaitKey] != ""
}
```

Reply:
Good catch :+1: ! not taking it just because needs to be re-worked out to support multiple services